### ISSUE TITLE
Allow dapr cli to run alone without app

### ISSUE THREAD

**Issue by RicardoNiepel** [dapr/cli#248]
## Describe the proposal

For debugging purpose it's currently necessary to use dummy commands as the app parameter for starting dapr side-by-side with the app process started already from the IDE.

The dummy commands need to different based on the OS.
For Linux and MacOS:
```
dapr run --app-id testapp --app-port 3000 --port 3500 --grpc-port 5001 -- cat
```
For Windows:
```
dapr run --app-id testapp --app-port 3000 --port 3500 --grpc-port 5001 -- waitfor FOREVER
```

**To make this scenario easier it would be better to allow dapr cli to run alone without app.**

See also: dapr/java-sdk#212

**Comment by philliphoff** [comment]
There are multiple platforms which do not allow debugging when forced to start the application via the `dapr` CLI, which then requires tooling to start `daprd` directly and mimic at least some of the behavior of the `dapr` CLI such as scaffolding the standalone components files.  Being able to start Dapr independently of the application (while reusing as much of the `dapr` CLI logic as possible) would be helpful.

How about something something like `dapr start` (as opposed to `dapr run`):

```bash
> dapr start --app-id my-app --app-port 5000
```

**Comment by yaron2** [comment]
> There are multiple platforms which do not allow debugging when forced to start the application via the `dapr` CLI, which then requires tooling to start `daprd` directly and mimic at least some of the behavior of the `dapr` CLI such as scaffolding the standalone components files. Being able to start Dapr independently of the application (while reusing as much of the `dapr` CLI logic as possible) would be helpful.
> 
> How about something something like `dapr start` (as opposed to `dapr run`):
> 
> ```shell
> > dapr start --app-id my-app --app-port 5000
> ```

I like `dapr start`, but would like to avoid adding new commands unless we absolutely have to.
Do you think it'd be natural to just allow `dapr run` without providing the app arguments? It seems natural to me; If I provide app arguments, Dapr will launch my app. If I don't, then only daprd will execute.

**Comment by bureado** [comment]
@yaron2 yes, `dapr run` was the first thing I tried and expected it to stay in the foreground with logs to `stdout`. Something like `dapr run -d` or `--detach` would mean to me run this as a service.

**Comment by philliphoff** [comment]
I'm fine with folding this into `dapr run`, too, but I think it needs an explicit flag (e.g. `--detached`, `-d` as suggested above) to help distinguish between the two cases.

**Comment by artursouza** [comment]
I've implemented the capability to run dapr without  app. I also tried the "-d" feature too (see PR).

I recommend we remove the detached option because it would require us to have logs persisted somewhere and rotated - it is a bigger scope (IMO) and not worth it. Users can simply invoke it with "&" and redirect stdout and stderr to their chosen destination.

See the PR on how to fork the process and how it only works on Unix-based system.
